src/gg.regression.unity.bots/Runtime/Scripts/ActionManager/RGActionProvider.cs

```diff
@@ -1,23 +1,88 @@
+    {
+        public RGSerializedAction[] actions;
+    }
+    
     /// <summary>
-    /// This interface provides the set of action types that were statically identified in the game.
-    /// This must be usable both in-editor and at run time.
+    /// This class reads and makes available the actions produced by RGActionAnalysis.
     /// </summary>
-    public interface IRGActionProvider
+    public class RGActionProvider
     {
-        /// <summary>
-        /// Event that is broadcasted when this provider's actions change.
-        /// </summary>
-        public event EventHandler ActionsChanged;
-        
+        public static readonly string ANALYSIS_RESULT_DIRECTORY = "Assets/Resources";
+        public static readonly string ANALYSIS_RESULT_NAME = "RGActionAnalysisResult";
+        public static readonly string ANALYSIS_RESULT_PATH = $"{ANALYSIS_RESULT_DIRECTORY}/{ANALYSIS_RESULT_NAME}.txt";
+
         /// <summary>
         /// Provides the static set of all action types identified in the game.
-        /// This is called once at the beginning of a bot/test session.
         /// </summary>
-        public IEnumerable<RGGameAction> Actions { get; }
+        public IEnumerable<RGGameAction> Actions => _actions;
+
+        public bool IsAvailable { get; private set; }
+
+        private IList<RGGameAction> _actions;
+
+        public RGActionProvider()
+        {
+            _actions = new List<RGGameAction>();
+            
+            var result = ReadAnalysisResult();
+            if (result != null)
+            {
+                try
+                {
+                    foreach (var serializedAction in result.actions)
+                    {
+                        _actions.Add(serializedAction.Deserialize());
+                    }
+
+                    IsAvailable = true;
+                }
+                catch (Exception e)
+                {
+                    RGDebug.LogWarning("Failed to read action analysis results (analysis needs to be re-run)\n" + e.StackTrace);
+                    _actions = new List<RGGameAction>();
+                    IsAvailable = false;
+                }
+            }
+            else
+            {
+                IsAvailable = false;
+            }
+        }
+
+        private RGActionAnalysisResult ReadAnalysisResult()
+        {
+            string jsonText = null;
+            #if UNITY_EDITOR
+            if (File.Exists(ANALYSIS_RESULT_PATH))
+            {
+                using (StreamReader sr = new StreamReader(ANALYSIS_RESULT_PATH))
+                {
+                    jsonText = sr.ReadToEnd();
+                }
+            }
+            #else
+            {
+                TextAsset jsonFile = Resources.Load<TextAsset>(ANALYSIS_RESULT_NAME);
+                jsonText = jsonFile?.text;
+            }
+            #endif
+            if (jsonText != null)
+            {
+                return JsonUtility.FromJson<RGActionAnalysisResult>(jsonText);
```

Comment:
I consider the action analysis output to be ephemeral data, since it needs to be re-generated whenever the game code changes. So, I don't really see the need to have backwards compatibility for the saved results format. If it substantially changes then we just re-run the analysis.

Reply:
It would be a nice user experience when they update the SDK to an incompatible version, to be able to tell them clearly that they need to regenerate their action data.

Reply:
I've added an API version field to RGActionAnalysisResult. If the current API version does not match the one that was parsed, then an exception is raised and the user will be requested to regenerate the action data.